kml.cc

```diff
@@ -448,15 +458,51 @@ void gx_trk_s(xg_string, const QXmlStreamAttributes*)
   track_add_head(gx_trk_head);
   delete gx_trk_times;
   gx_trk_times = new QList<gpsbabel::DateTime>;
+  delete gx_trk_coords;
+  gx_trk_coords = new QList<std::tuple<int, double, double, double>>;
 }
 
 void gx_trk_e(xg_string, const QXmlStreamAttributes*)
 {
+  // Check that for every temporal value (kml:when) in a kml:Track there is a position (kml:coord) value.
+  // Check that for every temporal value (kml:when) in a gx:Track there is a position (gx:coord) value.
+  if (gx_trk_times->size() != gx_trk_coords->size()) {
+    fatal(MYNAME ": There were more coord elements than the number of when elements.\n");
+  }
+
+  // In KML 2.3 kml:Track elements kml:coord and kml:when elements are not required to be in any order.
```

Comment:
As an aside, I don't think that was as much as conscious decision as much as it was difficult to explain it in the schema tools used.

Reply:
In general <sequence> was changed to <all> in the schema, moving the burden of ordering onto the reader.

KML 2.3 seems DOA, despite google prominently appearing among the submitters of the 2.3 OGC standard, and 3 years passing, GE is still using kml 2.2 and google extensions (xmlns:gx="http://www.google.com/kml/ext/2.2).  It was very hard to find any 2.3 samples.